lib/pages/home_page.dart

```diff
@@ -22,15 +22,56 @@ class Home extends StatelessWidget {
         preferredSize: Size.fromHeight(
           MeasuresConstants.appBarHeight,
         ),
-        child: Padding(
-          padding: const EdgeInsets.only(
-            top: MeasuresConstants.logoPaddingTop,
-          ),
-          child: Image.asset(
-            srcImage,
-            height: MeasuresConstants.logoHeight,
-            width: MeasuresConstants.logoWidth,
-          ),
+        child: Row(
+          children: [
+            Padding(
+              padding: const EdgeInsets.only(
+                top: MeasuresConstants.logoPaddingTop,
+              ),
+              child: Image.asset(
+                srcImage,
+                height: MeasuresConstants.logoHeight,
+                width: MeasuresConstants.logoWidth,
+              ),
+            ),
+            Padding(
+              padding: const EdgeInsets.only(
+                top: MeasuresConstants.textFieldPaddingTop,
+                left: MeasuresConstants.textFieldPaddingLeft,
+              ),
+              child: SizedBox(
+                child: TextField(
+                  style: TextStyle(
+                    color: Colors.white,
+                  ),
+                  decoration: InputDecoration(
+                    isDense: true,
+                    prefixIcon: Padding(
+                      padding: const EdgeInsets.only(right: 20.0),
+                      child: Icon(Icons.search, color: Colors.blue,),
+                    ),
+                    enabledBorder: UnderlineInputBorder(
+                      borderSide: BorderSide(
+                        color: Colors.blue,
+                      ),
+                    ),
+                  ),
+                  onSubmitted: (value) {
+                    switch (value){
+                      case "":
+                        bloc.fetchTrendingMovies();
+                        break;
+                      default:
+                        bloc.searchMovies(value);
+                        break;
+                    }
```

Comment:
This functionality can be inside Bloc, because here i dont want to know what you are doing with the logic of search. You can create a method inside Bloc like onSearchAction(value) or onSubmitted(value).

Reply:
Moved.